src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -113,6 +114,7 @@ internal static async Task InitializeStandardComponentServicesAsync(
         // It's important that this is initialized since a component might try to restore state during prerendering
         // (which will obviously not work, but should not fail)
         var componentApplicationLifetime = httpContext.RequestServices.GetRequiredService<ComponentStatePersistenceManager>();
+        componentApplicationLifetime.SetPlatformRenderMode(RenderMode.InteractiveAuto);
```

Comment:
I'm surprised by this being hardcoded to `RenderMode.InteractiveAuto`. Is there no reason to get the real value, whatever that would mean? Or if not, why is it even something that can vary?

Reply:
Actually I see it uses `InteractiveServer` in the non-endpoints cases below. Is the point of using `InteractiveAuto` that we don't know and are trying to support all cases here? Is there any value in being able to limit it to (say) Server even in the endpoint case if you know you're not using webassembly?

I'm guessing that whenever WebAssembly needs to be supported, the data has to be exposed in plaintext in a way that it doesn't if you're only using Server. Is that right?

Reply:
`Auto` does the right thing. We don't even know, so when we do `Auto` it renders to `Server` only if there are Server components, to Wasm only if there are Wasm components and to both if there are both on the page (or a component with `Auto`